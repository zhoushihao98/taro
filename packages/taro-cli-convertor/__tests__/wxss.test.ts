import Convertor from '../src/index'
import { clearMockFiles, getResMapFile, setMockFiles } from './__mocks__/fs-extra.js'
import { root } from './data/fileData'

const path = require('path')

jest.mock('path')

describe('wxss', () => {
  beforeAll(() => {
    jest.spyOn(Convertor.prototype, 'init').mockImplementation(() => {})
  })

  beforeEach(() => {
    // 清空文件信息
    clearMockFiles()
  })

  test('引入外部wxss', async () => {
    const wxssStr = `
      @import "../common.wxss";
      .scrollarea {
        flex: 1;
        overflow-y: hidden;
      }
    `
    const WITHOUT_WXSS_DEMO = {
      '/pages/common.wxss':`
        page {
          height: 100vh;
          display: flex;
          flex-direction: column;
          margin-top: 80px;
        }
      `,
      '/pages/index/index.wxss': wxssStr
    }
    setMockFiles(root, WITHOUT_WXSS_DEMO)
    const convertor = new Convertor(root, false)
    await convertor.traverseStyle(path.join(root,'/pages/index/index.wxss'), wxssStr)
    const resFileMap = getResMapFile()
    expect(resFileMap).toMatchSnapshot()
  })
})
