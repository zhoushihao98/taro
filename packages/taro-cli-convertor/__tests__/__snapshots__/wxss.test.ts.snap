// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wxss 引入外部wxss 1`] = `
Map {
  "/wxProject/taroConvert/src/pages/index/index.scss" => "
      @import "../common.scss";
      .scrollarea {
        flex: 1;
        overflow-y: hidden;
      }
    ",
}
`;
